### DISCUSSION TITLE
How to see executed SQL queries?

### DISCUSSION THREAD

**Discussion by wanjas** [prisma/prisma#3967] [Q&A] [answered]
Forgive me if I somehow missed this info.
Is there a way to log executed SQL queries on the client-side?

With `DEBUG=engine,prisma-client` there is plenty of data but no actual query. 
Right now the only option I see is to log incoming queries with the database itself which is not ideal. 

I really want to be able to run `ANALYZE` but in order to do so, I have to know what is actually executed.

**Comment by nikolasburk** [comment, marked as answer]
Hey @wanjas, thanks a lot for your question! 🙏 

Yes, you can log all generated SQL queries by setting the `log` option in the `PrismaClient` constructor:

```ts
const prisma = new PrismaClient({
  log: ['query'],
});
```

You can read more about this in the docs [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/logging) and [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/constructor#log) 🙂 Let me know if you have any further questions, always happy to help!

**Reply by wanjas** [reply]
Thank you a lot! 

I guess I should find some time to read the whole documentation once again already having some experience using Prisma.

**Reply by junajan** [reply]
Would be nice to have also way how to log only one particular query .. otherwise you can end up being flooded with too many logs

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
